client/metrics.go

```diff
@@ -0,0 +1,144 @@
+package client
+
+import (
+	"context"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/ory/fosite"
+	"github.com/ory/x/metricsx"
+)
+
+type Metrics struct {
+	prometheus.Collector
+	metricsx.Observer
+	Manager
+
+	Clients              prometheus.Gauge
+	ClientsCreated       prometheus.Counter
+	ClientsAuthenticated metricsx.CounterVec
+}
+
+func (m *Metrics) GetConcreteClient(ctx context.Context, id string) (*Client, error) {
+	return m.Manager.GetConcreteClient(ctx, id)
+}
+
+func (m *Metrics) GetClient(ctx context.Context, id string) (fosite.Client, error) {
+	return m.Manager.GetClient(ctx, id)
+}
+
+func (m *Metrics) UpdateClient(ctx context.Context, c *Client) error {
+	return m.Manager.UpdateClient(ctx, c)
+}
+
+func (m *Metrics) Authenticate(ctx context.Context, id string, secret []byte) (*Client, error) {
+	c, err := m.Manager.Authenticate(ctx, id, secret)
+	if err == nil {
+		m.ClientsAuthenticated.With(prometheus.Labels{
```

Comment:
Nice! This makes a ton of sense IMO. Here, "extending" the inner methods makes sense!

Reply:
Done! :)